### PR TITLE
Doc: Update Debian as Service Vm

### PR DESCRIPTION
**Pull request by fuzhongl**
Update steps from de-privileged boot to direct boot.

Signed-off-by: fuzhongl <fuzhong.liu@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: doc/tutorials/running_deb_as_serv_vm.rst [hunk 1]
```diff
@@ -44,88 +44,93 @@ Validated Versions
 ******************
 
 - **Debian version:** 10.0 (buster)
```
Comment (gvancuts) [inline doc/tutorials/running_deb_as_serv_vm.rst:46]: Can we update this? We mention Debian 10.0 but talk about 10.1 above in the doc and actually the download URL from the Debian download website only contains the 10.6 image. If that one (`Debian 10.6`) has been tested and validated, I'd recommend we use it throughout this doc.

File: doc/tutorials/running_deb_as_serv_vm.rst [hunk 2]
```diff
@@ -44,88 +44,93 @@ Validated Versions
...
-- **ACRN hypervisor tag:** acrn-2019w35.1-140000p
-- **Debian Service VM Kernel version:** 4.19.68-84.iot-lts2018-sos
+- **ACRN hypervisor tag:** acrn-2020w40.1-180000p
+- **Debian Service VM Kernel version:** 5.4.43-PKT-200203T060100Z
```
Comment (gvancuts) [inline doc/tutorials/running_deb_as_serv_vm.rst:48]: Further below, we instruct the use to clone and check out `release_2.2` from https://github.com/projectacrn/acrn-kernel, but that version does not match the tag you indicate here. Can you correct this by fixing the tag and/or the instructions to match what we have tested and validated?

File: doc/tutorials/running_deb_as_serv_vm.rst [hunk 3]
```diff
@@ -44,88 +44,93 @@ Validated Versions
...
+      Update the ``/etc/grub.d/40_custom`` file as shown below.
+
+   .. note::
+      Enter the command line for the kernel in ``/etc/grub.d/40_custom`` as
+      a single line and not as multiple lines. Otherwise, the kernel will
+      fail to boot.
+
+   .. code-block:: none
+
+      menuentry "ACRN Multiboot Gebian Service VM" --id debian-service-vm {
```
Comment (gvancuts) [inline doc/tutorials/running_deb_as_serv_vm.rst:96]:
```suggestion
      menuentry "ACRN Multiboot Debian Service VM" --id debian-service-vm {
```

File: doc/tutorials/running_deb_as_serv_vm.rst [hunk 4]
```diff
@@ -44,88 +44,93 @@ Validated Versions
...
+        search --no-floppy --fs-uuid --set 9bd58889-add7-410c-bdb7-1fbc2af9b0e1
+        echo 'loading ACRN...'
+        multiboot2 /boot/acrn/acrn.bin  root=PARTUUID="e515916d-aac4-4439-aaa0-33231a9f4d83"
+        module2 /boot/bzImage Linux_bzImage
+      }
+
+   .. note::
+      Update this to use the UUID (``--set``) and PARTUUID (``root=`` parameter)
+      (or use the device node directly) of the root partition (e.g.
+      ``/dev/nvme0n1p2). Hint: use ``sudo blkid /dev/sda*``.
```
Comment (gvancuts) [inline doc/tutorials/running_deb_as_serv_vm.rst:112]: You have problems with unclosed double-backticks here.

Can you also stick to one type of disk, either `/dev/sda` or `/dev/nvme0`?

File: doc/tutorials/running_deb_as_serv_vm.rst [hunk 5]
```diff
@@ -44,88 +44,93 @@ Validated Versions
...
       $ cd acrn-hypervisor
-      $ git checkout acrn-2019w35.1-140000p
+      $ git checkout acrn-2020w40.1-180000p
 
 #. Build and install ACRN:
 
    .. code-block:: bash
 
-      $ make BOARD=nuc7i7dnb FIRMWARE=uefi
+      $ make all BOARD_FILE=misc/vm_configs/xmls/board-xmls/whl-ipc-i7.xml SCENARIO_FILE=misc/vm_configs/xmls/config-xmls/whl-ipc-i7/industry.xml RELEASE=0
```
Comment (gvancuts) [inline doc/tutorials/running_deb_as_serv_vm.rst:66]: You are using a Whiskey Lake board here, but the prerequisite section still talks about a Kaby Lake NUC, can you update it?

**Review by gvancuts — CHANGES_REQUESTED** [review]

File: doc/tutorials/running_deb_as_serv_vm.rst (hunk 1 above)
Comment (fuzhongl) [reply doc/tutorials/running_deb_as_serv_vm.rst]: update to 10.1.

File: doc/tutorials/running_deb_as_serv_vm.rst (hunk 2 above)
Comment (fuzhongl) [reply doc/tutorials/running_deb_as_serv_vm.rst]: update with tag: ** acrn-2020w40.1-180000p

File: doc/tutorials/running_deb_as_serv_vm.rst (hunk 3 above)
Comment (fuzhongl) [reply doc/tutorials/running_deb_as_serv_vm.rst]: Done!

File: doc/tutorials/running_deb_as_serv_vm.rst (hunk 4 above)
Comment (fuzhongl) [reply doc/tutorials/running_deb_as_serv_vm.rst:112]: Sorry, not get your point.

File: doc/tutorials/running_deb_as_serv_vm.rst (hunk 5 above)
Comment (fuzhongl) [reply doc/tutorials/running_deb_as_serv_vm.rst]: update to Kaby Lake NUC

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: doc/tutorials/running_deb_as_serv_vm.rst [hunk 6]
```diff
@@ -43,89 +43,93 @@ Use the following instructions to install Debian.
 Validated Versions
 ******************
 
-- **Debian version:** 10.0 (buster)
-- **ACRN hypervisor tag:** acrn-2019w35.1-140000p
-- **Debian Service VM Kernel version:** 4.19.68-84.iot-lts2018-sos
+- **Debian version:** 10.1 (buster)
+- **ACRN hypervisor tag:** acrn-2020w40.1-180000p
+- **Debian Service VM Kernel version: tag:** acrn-2020w40.1-180000p
```
Comment (gvancuts) [inline doc/tutorials/running_deb_as_serv_vm.rst:48]: I would prefer that we stay consistent with the instructions below, so either we mention the branch (`release_2.2`) here, or we change the instructions below to check out this tag (and not the branch).

File: doc/tutorials/running_deb_as_serv_vm.rst [hunk 7]
```diff
@@ -43,89 +43,93 @@ Use the following instructions to install Debian.
 Validated Versions
 ******************
 
-- **Debian version:** 10.0 (buster)
-- **ACRN hypervisor tag:** acrn-2019w35.1-140000p
-- **Debian Service VM Kernel version:** 4.19.68-84.iot-lts2018-sos
+- **Debian version:** 10.1 (buster)
```
Comment (gvancuts) [inline doc/tutorials/running_deb_as_serv_vm.rst:46]: The only ISO available from the download area is 10.6. Can we use that here? (In other words, have we validated it?)

File: doc/tutorials/running_deb_as_serv_vm.rst [hunk 8]
```diff
@@ -43,89 +43,93 @@ Use the following instructions to install Debian.
...
+        search --no-floppy --fs-uuid --set 9bd58889-add7-410c-bdb7-1fbc2af9b0e1
+        echo 'loading ACRN...'
+        multiboot2 /boot/acrn/acrn.bin  root=PARTUUID="e515916d-aac4-4439-aaa0-33231a9f4d83"
+        module2 /boot/bzImage Linux_bzImage
+      }
+
+   .. note::
+      Update this to use the UUID (``--set``) and PARTUUID (``root=`` parameter)
+      (or use the device node directly) of the root partition (e.g.
+      ``/dev/nvme0n1p2``). Hint: use ``sudo blkid /dev/sda*``.
```
Comment (gvancuts) [inline doc/tutorials/running_deb_as_serv_vm.rst:112]: What I mean is that the first sentence uses an NVMe drive for the rootfs. But the second sentence where we give the hint assumes a standard SATA disk, it's not a big deal but a bit confusing. How about this?
```suggestion
      ``/dev/nvme0n1p2``). Hint: use ``sudo blkid <device node>``.
```

File: doc/tutorials/running_deb_as_serv_vm.rst (hunk 7 above)
Comment (fuzhongl) [reply doc/tutorials/running_deb_as_serv_vm.rst:46]: No bandwidth to verify 10.6.
@NanlinXie  Any direction?
Thanks!

File: doc/tutorials/running_deb_as_serv_vm.rst (hunk 6 above)
Comment (fuzhongl) [reply doc/tutorials/running_deb_as_serv_vm.rst]: Done!

File: doc/tutorials/running_deb_as_serv_vm.rst (hunk 8 above)
Comment (fuzhongl) [reply doc/tutorials/running_deb_as_serv_vm.rst]: Done!

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: doc/tutorials/running_deb_as_serv_vm.rst (hunk 7 above)
Comment (NanlinXie) [reply doc/tutorials/running_deb_as_serv_vm.rst:46]: @gvancuts , our previous validation is based on 10.1 version. I have created one git issue to track this.
https://github.com/projectacrn/acrn-hypervisor/issues/5408

File: doc/tutorials/running_deb_as_serv_vm.rst [hunk 9]
```diff
@@ -43,89 +43,94 @@ Use the following instructions to install Debian.
...
-         Boot0006* UEFI : LAN : PXE IP6 Intel(R) Ethernet Connection I219-LM  PciRoot(0x0)/Pci(0x1f,0x6)/MAC(54b2030f4b84,0)/IPv6([::]:<->[::]:,0,0)..BO
-         Boot0007* UEFI : Built-in EFI Shell     VenMedia(5023b95c-db26-429b-a648-bd47664c8012)..BO
-         Boot0008* Linux bootloader VenHw(99e275e7-75a0-4b37-a2e6-c5385e6c00cb)
-         Boot0009* ACRN Hypervisor HD(1,GPT,94597852-7166-4216-b0f1-cef5fd1f2349,0x800,0x100000)/File(\EFI\acrn\acrn.efi)b.o.o.t.l.o.a.d.e.r.=.\.E.F.I.\.d.e.b.i.a.n.\.g.r.u.b.x.6.4...e.f.i.
-
-   #. Install the Service VM kernel and reboot:
+         $ mkdir ~/sos-kernel && cd ~/sos-kernel
+         $ git clone https://github.com/projectacrn/acrn-kernel
+         $ cd acrn-kernel
+         $ git checkout -b v2.2 remotes/origin/release_2.2
```
Comment (gvancuts) [inline doc/tutorials/running_deb_as_serv_vm.rst:78]: Let's make this a little easier:

```suggestion
         $ git checkout release_2.2
```

File: doc/tutorials/running_deb_as_serv_vm.rst (hunk 7 above)
Comment (gvancuts) [reply doc/tutorials/running_deb_as_serv_vm.rst:46]: Sounds good, let's discuss how to tackle this problem in there.

**Review by gvancuts — APPROVED** [review]
One minor change requested but this is now good to go!

**Comment by wenlingz** [comment]
@fuzhongl make doc fail, please refer:
****************************************************
make[1]: Entering directory '/root/workspace/acrn-hypervisor-premerge-build-image4/acrn-hypervisor/doc'

New errors/warnings found, please fix them:
==============================================

/root/workspace/acrn-hypervisor-premerge-build-image4/acrn-hypervisor/build/doc/rst/tutorials/running_deb_as_serv_vm.rst:48: WARNING: Inline strong start-string without end-string.

make[1]: Leaving directory '/root/workspace/acrn-hypervisor-premerge-build-image4/acrn-hypervisor/doc'
make[1]: *** [Makefile:61: html] Error 1
make: *** [Makefile:184: doc] Error 2
++ upload /root/workspace/acrn-hypervisor-premerge-build-image4/artifact/5404 sit-image.sh.intel.com /home/sit/workspace/images/pme-build/acrn-hypervisor sit 123456
++ [[ 5 -ne 5 ]]
++ local src=/root/workspace/acrn-hypervisor-premerge-build-image4/artifact/5404
++ local ip=sit-image.sh.intel.com

File: doc/tutorials/running_deb_as_serv_vm.rst (hunk 9 above)
Comment (fuzhongl) [reply doc/tutorials/running_deb_as_serv_vm.rst]: Done!

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: doc/tutorials/running_deb_as_serv_vm.rst [hunk 10]
```diff
@@ -137,16 +142,13 @@ Install ACRN on the Debian VM
...
 
       $ sudo systemctl enable systemd-networkd
       $ sudo systemctl start systemd-networkd
 
-#. Prepare and Start a User VM.
-
-   .. important:: Need instructions for this.
+Prepare and Start a User VM
+***************************
+.. important:: Need instructions for this.
```
Comment (dbkinder) [inline doc/tutorials/running_deb_as_serv_vm.rst:154]: We're still missing the instructions for preparing and starting a User VM.  Can you add it with this PR?

**Review by dbkinder — CHANGES_REQUESTED** [review]
can you add the missing instructions in this PR?

File: doc/tutorials/running_deb_as_serv_vm.rst (hunk 10 above)
Comment (fuzhongl) [reply doc/tutorials/running_deb_as_serv_vm.rst:154]: @dbkinder 
**sudo make install** has copied acrn-dm and OVMF to the target directory.
And the correct IASL version was also build at previous step.
There is no more steps for  User VM setup.
Can we remove this item?
Thanks!

Comment (gvancuts) [reply doc/tutorials/running_deb_as_serv_vm.rst:154]: I'm OK with removing this, or add a note to indicate the users can refer to our other documents on how to prepare a User VM image.

Comment (gvancuts) [reply doc/tutorials/running_deb_as_serv_vm.rst:154]: > If there are indeed no more steps needed, then yes, you can update this PR and delete this heading and "important" note. @gvancuts do you agree?

Yes, please remove this @fuzhongl

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Review by gvancuts — APPROVED** [review]
Thanks!

**Review by dbkinder — APPROVED** [review]

**Pull request merged**